### ISSUE TITLE
Unexpected error when negate reference to number

### ISSUE THREAD

**Issue by p0lunin** [intellij-rust/intellij-rust#5765]
## Environment

* **IntelliJ Rust plugin version:** 0.3.126.3220-201
* **Rust toolchain version:** 1.44.0 (49cae5576 2020-06-01) x86_64-pc-windows-msvc
* **IDE name and version:** PyCharm 2020.1.2 Community Edition (PC-201.7846.77)
* **Operating system:** Windows 10 10.0

## Problem description

Unexpected error when negate reference to number.

## Steps to reproduce
```rust
fn foo() {
    let num = &10;
    let num2: i32 = -num;
}
```
